apps/obsidian/src/components/TldrawView.tsx

```diff
@@ -102,23 +115,32 @@ export class TldrawView extends TextFileView {
     }
   }
 
+  private assertInitialized(): void {
+    if (!this.file) throw new Error("TldrawView not initialized: missing file");
+    if (!this.assetStore)
+      throw new Error("TldrawView not initialized: missing assetStore");
+    if (!this.store)
+      throw new Error("TldrawView not initialized: missing store");
+  }
+
   private createReactRoot(entryPoint: Element, store: TLStore) {
     const root = createRoot(entryPoint);
-    if (!this.file) return;
+    this.assertInitialized();
 
     root.render(
       <React.StrictMode>
         <TldrawPreviewComponent
           store={store}
           plugin={this.plugin}
-          file={this.file}
+          file={this.file!}
```

Comment:
We should avoid `!` whenever possible.  It basically just bypasses type safety.

If you inline the `assertInitialized()`, the `!` is not necessary.